### PR TITLE
Add Partner Register

### PR DESCRIPTION
**Pull request by VincentGJ1**

### CODE REVIEW

File: src/partner/entities/company/address/address.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
+import { AbstractEntity } from '../../../../abstractEntity';
+
+export enum AddressFields {
+  COLUMN_LINE1 = 'line1',
+  COLUMN_LINE2 = 'line2',
+  COLUMN_CITY = 'city',
+  COLUMN_STATE = 'state',
+  COLUMN_ZIP = 'zip',
+  COUNTRY_CODE = 'countryCode',
```
Comment (bdjonkamla) [inline src/partner/entities/company/address/address.ts:9]: `COLUMN_COUNTRY_CODE = 'countryCode',`

File: src/partner/entities/company/address/address.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+export type AddressType = {
+  [AddressFields.COLUMN_LINE1]: string;
+  [AddressFields.COLUMN_LINE2]: string;
+  [AddressFields.COLUMN_CITY]: string;
+  [AddressFields.COLUMN_STATE]: string;
+  [AddressFields.COLUMN_ZIP]: string;
+  [AddressFields.COUNTRY_CODE]: string;
+};
```
Comment (bdjonkamla) [inline src/partner/entities/company/address/address.ts:19]: All fields are mandatory ?

File: src/partner/entities/company/company.ts [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+  internalReference: string;
+  phone: string;
+  numberOfEmployee: string;
+  corporateEmail: string;
+  website: string;
+  vatNumber: string;
+  registrationNumber: string;
+  address: AddressType;
+  billingAddress: AddressType;
+};
```
Comment (bdjonkamla) [inline src/partner/entities/company/company.ts:30]: All fields are mandatory ?

File: src/partner/entities/contact/contact.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+export type PartnerContactType = {
+  [PartnerContactFields.COLUMN_TITLE]: string;
+  [PartnerContactFields.COLUMN_FIRSTNAME]: string;
+  [PartnerContactFields.COLUMN_LASTNAME]: string;
+  [PartnerContactFields.COLUMN_PHONE]: string;
+  [PartnerContactFields.COLUMN_EMAIL]: string;
+  [PartnerContactFields.COLUMN_TYPE]: string;
+  [PartnerContactFields.COLUMN_COUNTRY]: string;
+};
```
Comment (bdjonkamla) [inline src/partner/entities/contact/contact.ts:21]: All fields are mandatory ?

File: src/partner/entities/dataPartner.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  COLUMN_USER = 'user',
+  COLUMN_CONTACT = 'contact',
+  COLUMN_COMPANY = 'company',
+}
+
+export type DataPartnerType = {
+  [DataPartnerFields.COLUMN_USER]: UserType;
+  [DataPartnerFields.COLUMN_CONTACT]: ContactType;
+  [DataPartnerFields.COLUMN_COMPANY]: CompanyType;
+};
```
Comment (bdjonkamla) [inline src/partner/entities/dataPartner.ts:22]: All fields are mandatory ?

**Review by fdianoux — APPROVED** [review]

**Review by BlondJP — APPROVED** [review]

File: src/partner/entities/company/company.ts [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+  COLUMN_CORPORATE_EMAIL = 'corporateEmail',
+  COLUMN_WEBSITE = 'website',
+  COLUMN_VAT_NUMBER = 'vatNumber',
+  COLUMN_REGISTRATION_NUMBER = 'registrationNumber',
+  COLUMN_ADDRESS = 'address',
+  COLUMN_BILLING_ADDRESS = 'billingAddress',
+}
+
+export type PartnerCompanyType = {
+  name: string;
```
Comment (fdianoux) [inline src/partner/entities/company/company.ts:19]:
```suggestion
  [PartnerCompanyFields.COLUMN_NAME]: string;
```

**Review by bdjonkamla — APPROVED** [review]

**Pull request merged**